[zh] Localize 'agile_software_development'

Anddd7 opened the pull request:
Localize 'agile_software_development'. (hah, a very familiar topic, in my daily work)

fix #849

Rocksnake commented:
I suggest you split feature and fix into two tasks next time. Rules for keeping a PR that only covers one thing.
1. feature: Localize 'agile_software_development'. (hah, a very familiar topic, in my daily work)
2. fix: fixed some inconsistent subheadings

Anddd7 commented:
> I suggest you split feature and fix into two tasks next time. Rules for keeping a PR that only covers one thing.
> 
> 1. feature: Localize 'agile_software_development'. (hah, a very familiar topic, in my daily work)
> 2. fix: fixed some inconsistent subheadings

Thanks for your suggestions, split the subheading fix into https://github.com/cncf/glossary/pull/916 and revised content.

Jacob953 reviewed (approved):
LGTM.
